sqlmesh/utils/file.py

```diff
@@ -0,0 +1,124 @@
+import json
+import random
+import time
+import typing as t
+from contextlib import contextmanager
+from uuid import uuid4
+
+import fsspec
+
+class FileTransactionHandler:
+    """A transaction handler which wraps a file path."""
+
+    def __init__(self, path: str) -> None:
+        """Creates a new FileTransactionHandler.
+        
+        Args:
+            path: The path to lock. It should follow the format of a table name.
+                Something like snapshots/version=<ver>/<snapshot_id>
+        """
+        self.path = path
+        # TODO: pass in a filesystem, naturally
+        self._fs: fsspec.AbstractFileSystem = fsspec.filesystem("file")
+        self._original_contents: t.Optional[bytes] = None
+        self._is_locked = False
+
+    @property
+    def lock_path(self) -> str:
+        """The key which represents a path to the file's lock."""
+        return f"{self.path}.lock"
+
+    def read(self) -> t.Optional[bytes]:
```

Comment:
in state sync, we call it lock_for_update, which means the read acquires a lock

Reply:
i see, you already have read_with_lock, that's fine, in that case we don't need the check on line 34, if not self._is_locked